torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -6038,37 +6042,70 @@ def test_cuda_future_device_not_cuda(self):
         ):
             fut = Future(devices=["cpu"])
 
-    def _test_cuda_future_extraction(self, wrapper, unwrapper):
-        # We check proper CUDA stream synchronization by filling the tensor with
-        # the expected value in one stream, and reading it from another stream.
-        tensor = torch.zeros((100,), device="cuda:0")
+    def _test_cuda_future_extraction(self, wrapper, unwrapper, sparse_tensor):
+        # We check proper CUDA stream synchronization by adding to the tensor
+        # in one stream to get the expected value, and reading it from another stream.
+        if sparse_tensor:
+            tensor = build_sparse_tensor().to("cuda:0")
+            add_tensor = build_sparse_tensor().to("cuda:0")
+            expected_tensor = (tensor + add_tensor).coalesce()
+        else:
+            tensor = torch.zeros((100,), device="cuda:0")
+            add_tensor = torch.ones((100,), device="cuda:0")
+            expected_tensor = tensor + add_tensor
```

Comment:
This is a preexisting issue, but I'm realizing now that probably we shouldn't create the tensors in the default (unset) stream and then use them in another one without explicitly synchronizing. Perhaps we could in fact move this whole creation step inside `with torch.cuda.stream(stream):` (but before the `torch.cuda._sleep()` call)?

Reply:
okay update the code.